src/site/content/en/learn/pwa/serving/index.md

```diff
@@ -1,11 +1,242 @@
+
+The `match()` function receives an HTTP request or a URL as an argument and it returns a promise that resolves with the Response associated with the corresponding key.
+
+```js
+// Global search on all caches in the current origin
+caches.match(urlOrRequest).then(response => {
+   console.log(response ? response : "It's not in the cache");
+});
+
+// Cache-specific search
+caches.open("pwa-assets").then(cache => {
+  cache.match(urlOrRequest).then(response) {
+    console.log(response ? response : "It's not in the cache");
+  }
+});
+```
+
+{% Aside 'gotchas' %}
+When using `match()` with `respondWith()`, the match promise will resolve either with the found item or with nothing. It will only reject if there was a problem that prevented the search operation from completing. To throw an error if no results are found, you'll need to inspect the results before returning a response to `respondWith()`.
+{% endAside %}
+
+## Caching strategies
+
+Serving files only from the browser cache doesn't fit every use case. For example the user or the browser can evict the cache. That's why you should define your own strategies for delivering assets for your PWA.
+You're not restricted to one caching strategy. You can define different ones for different URL patterns. For example, you can have one strategy for the minimum UI assets, another for API calls, and a third for image and data URLs.
+To do this, read `event.request.url` in `ServiceWorkerGlobalScope.onfetch` and parse it through regular expressions or a [URL Pattern](/urlpattern/). (At the time of writing, URL Pattern is not supported on all platforms).
+
+The most common strategies are:
+
+Cache First
+: Searches for a cached response first and falls back to the network if one isn't found.
+
+Network First
+: Requests a response from the network first and if none is returned, checks for response in the cache.
+
+Stale While Revalidate
+: Serves a response from the cache, while in the background requests the latest version and saves it to the cache for the next time the asset is requested.
+
+Network-Only
+: Always replies with a response from the network or errors out. The cache is never consulted.
+
+Cache-Only
+: Always replies with a response from the cache or errors out. The network will never be consulted. The assets that will be served using this strategy must be added to the cache before they are requested.
+
+### Cache first
+
+Using this strategy, the service worker looks for the matching request in the cache and returns and returns the corresponding Response if it's cached. Otherwise it retrieves the response from the network (optionally, updating the cache for future calls). If there is neither a cache response nor a network response, the request will error. Since serving assets without going to the network tends to be faster, this strategy prioritizes performance over freshness.
```

Comment:
```suggestion
Using this strategy, the service worker looks for the matching request in the cache and returns the corresponding Response if it's cached. Otherwise it retrieves the response from the network (optionally, updating the cache for future calls). If there is neither a cache response nor a network response, the request will error. Since serving assets without going to the network tends to be faster, this strategy prioritizes performance over freshness.
```